grc/core/blocks/embedded_python.py

```diff
@@ -149,10 +149,10 @@ def rewrite(self):
     def _update_params(self, params_in_src):
         param_factory = self.parent_platform.make_param
         params = {}
-        for param in list(self.params):
-            if hasattr(param, '__epy_param__'):
-                params[param.key] = param
-                del self.params[param.key]
+        for key, value in self.params.copy().items():
```

Comment:
Do we know if this works with Python 2?

Reply:
true, I should have tested the thing under Py2k; don't even have a system at hand to do that, and no test coverage of that, so can't do testing on CI...